mne/datasets/utils.py

```diff
@@ -252,6 +252,8 @@ def _download_mne_dataset(
         download=download,
         accept=accept,
     )
+    assert isinstance(path, Path)
+    return path
```

Comment:
Is this really necessary? For me, the return type annotation is sufficient for the type checker to infer the correct return type.

Reply:
Yes but without this type guard, the return value here in this fu drin will get red squiggly lines. This is of course not user-facing but something that drives me mad as a developer, so I added the assert statement to take care of it